### ISSUE TITLE
SOW 2 - Task 2.3 - Consistency of Imputed Catch

### ISSUE THREAD

**Issue by jasmyace** [tmcd82070/CAMP_RST#25]
Revisit and refine how R is calculating imputed catch in the production report's daily base files so the imputed values do not produce values that appear to be markedly out-of-line with adjoining days when catch was not imputed.

**Comment by jasmyace** [comment]
(from Doug)

The JPEG below illustrates this issue.  Felipe in my office told me last year there is a setting in R that can be set to limit how carried away R gets when it imputes a value.  I will catch Felipe in the next day or two to see what he was talking about, and then I will send you and email giving you a lead based on what he knows so you can work more quickly.

![image002](https://cloud.githubusercontent.com/assets/9057972/11961296/e42d0ab0-a894-11e5-8911-c04a8d32d47b.png)

**Comment by jasmyace** [comment]
I'm trying to reproduce this output, in order to see what the spline plot looks like.  (It's worth noting, however, that the plot above is for passage, and so possibly, variable-over-time efficiency may be playing a role.) 

In any case, does "juvenile" mean fry?  Was this output from the "ALL Runs" report, or the "lifeStage" one?

**Comment by jasmyace** [comment]
In my work on Task 2.1, I'm working on removing the preceding and antecedent zeros for runs on the American.  I can clearly see that this is Fall run for Fry.  Thanks.

**Comment by dougthreloff** [comment]
Jason:



I couldn’t find my original file that identified the issue with task 2.3,
but I was able to re-create the issue today by running a new All Runs
report.  See the attached file, and read the readme file first.



Doug



*From:* Jason [mailto:notifications@github.com]
*Sent:* Wednesday, January 20, 2016 7:47 AM
*To:* tmcd82070/CAMP_RST
*Subject:* Re: [CAMP_RST] SOW 2 - Task 2.3: Consistency of Imputed Catch
(#25)



I'm trying to reproduce this output, in order to see what the spline plot
looks like. (It's worth noting, however, that the plot above is for
passage, and so possibly, variable-over-time efficiency may be playing a
role.)

In any case, does "juvenile" mean fry? Was this output from the "ALL Runs"
report, or the "lifeStage" one?

—
Reply to this email directly or view it on GitHub
<https://github.com/tmcd82070/CAMP_RST/issues/25#issuecomment-173244708>.[image:
Image removed by sender.]

**Comment by jasmyace** [comment]
Task 2.1 required a large amount of code to be run.  One of the products it produced was the spline plot for this particular data sequence of catch, before and after the zero deletion update.  Although my time frames are slightly different than what Doug used above, comparing the dot-pattern between these and Doug's communicates the two are sufficient similar to move the discussion along.  In both, note that the not-great fit is imputing a lot more fish than might be desired in the beginning of February, given the low observed fish count on Feb. 8th. The issue here is that the low true observed count of Feb. 8th should be serving as more of an anchor to the resulting spline than it appears to be.  It's also worth noting that Doug plotted passage, while I plotted catch.  However, in both, the issue is clear.  

**Before Zero Deletion**
![run 2--year_american river_american river at watt avenue_2013-12-31_2014-05-30_north channel 8 1_fall_fry_spline](https://cloud.githubusercontent.com/assets/9057972/12524146/4e6cb57e-c118-11e5-86d5-58fb91a1c086.png)

I had secretly hoped that in dealing with all of those trailing zeros, the fit would be better.  I believe that it was only 2013 where the gear swap-out occurred, so I think all of those days of zero Fall Fry are legit.  Here is how the plot looks when the preceding and trailing zeros are removed.  

**After Zero Deletion**
![run 2--year_american river_american river at watt avenue_2013-12-31_2014-05-30_north channel 8 1_fall_fry_spline](https://cloud.githubusercontent.com/assets/9057972/12524284/3eba294e-c119-11e5-87fe-b53dfa8272ba.png)

Note that zero deletion didn't change anything in the recorded data at the start of the season -- you can infer this from the pattern of two black dots, followed by the hollow blue square of imputation on day three, when comparing the before and after.  But, the deletion of all the trailing zeros is apparent.  Observe the scaling of the y-axis;  black dots that *look* like zero catch may simply be small counts.  For example, the last day in the new plot (April 10th) caught 9 Fall fry (1 assigned, and 8 via plus-counting).  

In looking at the data in this plot, it seems clear that there is a pulse of fish.  In fact, in this case, it seems that zero deletion made things worse.  At least before, the obvious pulse of fish was captured by the spline.  However, in looking at this plot more closely, I see that this sequence of observed catch is different in most others that I have seen.  I first saw this by observing the local minimum around Feb. 26th, and then the pulse of fish before this day, and the second pulse after this day.  I don't think I have seen two separate pulses of fish that have occurred in such a short time frame.  So, I would want the spline plot to look something like this:  

![image](https://cloud.githubusercontent.com/assets/9057972/12524946/026881fc-c11e-11e5-8153-ebff69c4f46c.png)

Here, it seems plausible to assume that two pulses have occurred.  However, I suspect that the current setup of the code would prevent the use of a spline curve this complicated, with two maxima, three minima (including the endpoints), and four inflection points, in such a short time frame.  It could also be the case that the biologists would never agree that a double-pulse could take place, and this is just a data nuance.  

Assuming that this is a double-signal however, we could work to achieve a better fit.  We have discussed before the use of a smoothing parameter within the spline paradigm;  change it, and the spline could follow the data more readily.  However, it's only this plot that has the issue -- not all of them, on all lifestages, runs, and rivers.  So, changing the default value/methodology used may do more harm than good.  

Given the poor fit of the underlying spline in this case, I think it makes sense to consider the possibility that some curves may need a bit more allowed "wiggle" to capture their true observed pattern of catch than others.  In thinking about it, it makes sense to me that different rivers and runs and lifestages could, on average, expect different amounts of wiggle.  It might be worthwhile to consider the possibility that different selection criteria should lead to different amounts of wiggle.  

So, in other words, whenever imputed values are "markedly out of line" with respect to observed counts of fish, maybe we need to develop a methodology to allow for better spline fitting.  This could take place via a data-driven redefining of the smoothing parameter, or "wiggle."  In this vein, I think that the spline plots will help us to identify which fits are better than others, and from which, we could then hypothesize ways of moving forward.
